### PR TITLE
Search academy API as well as the database and compare results

### PR DESCRIPTION
**Pull request by elcorbs**
When searching Academy, search both the DB and API then log the comparison between results but still return the results from the DB.

### CODE REVIEW

**Comment by bjpirt** [comment]
Had a cursory glance through this and it looks good to me - wonder if @mmmmillar or one of the HN team might be able to take a look through this too?

File: api/lib/gateways/Academy-Benefits/Search.js [hunk 1]
```diff
@@ -1,80 +1,76 @@
...
 
-  const validateIds = record => {
-    return record.claim_id && record.check_digit && record.person_ref;
-  };
+  const sameResults = (dbRecords, apiRecords) => {
+    if (dbRecords.length !== apiRecords.length) return false;
+    let equal = true;
+    dbRecords.forEach(record => {
+      if (!apiRecords.find(r => recordEquality(record, r))) {
+        equal = false;
```
Comment (mmmmillar) [inline api/lib/gateways/Academy-Benefits/Search.js:47]: can equal be set to true again here?

Comment (elcorbs) [reply api/lib/gateways/Academy-Benefits/Search.js:47]: I'm not sure I know what you mean. 
The aim is for it to go through each record returned from the database and check that it has an equivalent record from the API and if not to set equal to false. If every record is found the API records, the equal variable will remain true. - Does that make sense?

File: api/lib/gateways/Academy-Benefits/Search.js [hunk 2]
```diff
@@ -1,80 +1,76 @@
...
+    return record1.id === record2.id
+      && stringDataEquality(record1.firstName, record2.firstName)
+      && stringDataEquality(record1.lastName, record2.lastName)
+      && stringDataEquality(record1.nino, record2.nino)
+      && stringDataEquality(record1.postcode, record2.postcode)
+      && (record1.dob === record2.dob || moment(record1.dob).diff(moment(record2.dob), 'days') < 1)
+  }
+
+  const stringDataEquality = (string1, string2) => {
+    if (string1 == string2) return true;
```
[the remark this answers is not in the document]
Comment (elcorbs) [reply api/lib/gateways/Academy-Benefits/Search.js:63]: Yeah, probably could've been more explicit with that, but this felt more succinct.

File: api/lib/gateways/Academy-Benefits/Search.js (hunk 1 above)
Comment (mmmmillar) [reply api/lib/gateways/Academy-Benefits/Search.js:47]: Yeah so if the first record is not is apiRecords, equal is set to false. But then if the second record is apiRecords, equal will get set to true again? Or am I reading it wrong? I think it maybe needs a break?

Comment (elcorbs) [reply api/lib/gateways/Academy-Benefits/Search.js:47]: It won't reset it to true, it'll just do nothing. But equally a break does make sense, no point continuing the loop if it already failed so I'll add it in

Comment (mmmmillar) [reply api/lib/gateways/Academy-Benefits/Search.js:47]: Haha, never mind me!

**Pull request merged**